pkg/reconciler/pullsubscription/pullsubscription.go

```diff
@@ -191,21 +193,28 @@ func (c *Reconciler) reconcile(ctx context.Context, source *v1alpha1.PullSubscri
 		return nil
 	}
 
-	sinkURI, err := duck.GetSinkURI(ctx, c.DynamicClientSet, source.Spec.Sink, source.Namespace)
-	if err != nil {
-		source.Status.MarkNoSink("NotFound", "")
-		return err
+	// Sink is required. // TODO: this is a problem for channels.
```

Comment:
why is this a problem for channels? Is there something that needs to be done for channels? perhaps create an issue for it?

Reply:
I was wrong, I have a plan. Todo removed.